pkg/agent/agent.go

```diff
@@ -165,16 +167,19 @@ func (a *Agent) setupProfiling(ctx context.Context) (stop func()) {
 
 func (a *Agent) attest(ctx context.Context, cat catalog.Catalog, metrics telemetry.Metrics) (*attestor.AttestationResult, error) {
 	config := attestor.Config{
-		Catalog:           cat,
-		Metrics:           metrics,
-		JoinToken:         a.c.JoinToken,
-		TrustDomain:       a.c.TrustDomain,
-		TrustBundle:       a.c.TrustBundle,
-		InsecureBootstrap: a.c.InsecureBootstrap,
-		BundleCachePath:   a.bundleCachePath(),
-		SVIDCachePath:     a.agentSVIDPath(),
-		Log:               a.c.Log.WithField(telemetry.SubsystemName, telemetry.Attestor),
-		ServerAddress:     a.c.ServerAddress,
+		Catalog:                cat,
+		Metrics:                metrics,
+		JoinToken:              a.c.JoinToken,
+		TrustDomain:            a.c.TrustDomain,
+		TrustBundle:            a.c.TrustBundle,
+		InsecureBootstrap:      a.c.InsecureBootstrap,
+		BundleCachePath:        a.bundleCachePath(),
+		SVIDCachePath:          a.agentSVIDPath(),
+		Log:                    a.c.Log.WithField(telemetry.SubsystemName, telemetry.Attestor),
+		ServerAddress:          a.c.ServerAddress,
+		ExperimentalAPIEnabled: a.c.ExperimentalAPIEnabled,
+		CreateNewAgentClient:   agent.NewAgentClient,
+		CreateNewBundleClient:  bundle.NewBundleClient,
```

Comment:
nit: one thing we can do to make this harder to misconfigure is to have the attestor default to `agent.NewAgentClient` and `bundle.NewBundleClient` if these functions are unset in the configuration.